rust/polars-ocaml-macros/src/lib.rs

```diff
@@ -123,6 +132,74 @@ pub fn ocaml_interop_export(_input: TokenStream, annotated_item: TokenStream) ->
     TokenStream::from(expanded)
 }
 
+// `std::panic::catch_unwind` only stores the error string passed to the panic,
+// which means that we can't figure out the backtrace by the time we see the Err
+// case. So, we use `std::panic::set_hook` to store the backtrace in a thread
+// local variable to be retrieved when the unwind is caught as discussed here:
+// https://stackoverflow.com/a/73711057
+#[proc_macro]
+pub fn ocaml_interop_backtrace_support(_item: TokenStream) -> TokenStream {
+    let expanded = quote! {
+        thread_local! {
+            static LAST_BACKTRACE: ::std::cell::RefCell<Option<::std::backtrace::Backtrace>> =
+                ::std::cell::RefCell::new(None);
```

Comment:
For this usage, `Cell` is enough.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c5f596f8336304cb7bb31edc64a85a73